### ISSUE TITLE
Counter id is generated using DeprecatedTags and ignoring Tags

### ISSUE THREAD

**Issue by kitemongerer** [cloudfoundry/loggregator-agent#9]
If I emit counters with different tags, I expect their totals to be tracked separately. However, this functionality only exists if I am using the DeprecatedTags field.

This line is the issue: https://github.com/cloudfoundry/loggregator-agent/blob/77da6d528025d41c23392254b6f26e31e821a735/pkg/egress/v2/counteraggregator.go#L39

Failing test made based on existing test by replacing DeprecatedTags with Tags:

```
    It("calculates totals separately for counter envelopes with same name but unique tags", func() {
        mockWriter := newMockWriter()
        close(mockWriter.WriteOutput.Ret0)

        aggregator := egress.NewCounterAggregator(mockWriter)
        aggregator.Write([]*loggregator_v2.Envelope{{
            Message: &loggregator_v2.Envelope_Counter{
                Counter: &loggregator_v2.Counter{
                    Name:  "name",
                    Delta: 10,
                },
            },
            Tags: map[string]string{
                "code": "200",
            },
        }})
        aggregator.Write([]*loggregator_v2.Envelope{{
            Message: &loggregator_v2.Envelope_Counter{
                Counter: &loggregator_v2.Counter{
                    Name:  "name",
                    Delta: 15,
                },
            },
            Tags: map[string]string{
                "code": "200",
            },
        }})
        aggregator.Write([]*loggregator_v2.Envelope{{
            Message: &loggregator_v2.Envelope_Counter{
                Counter: &loggregator_v2.Counter{
                    Name:  "name",
                    Delta: 20,
                },
            },
            Tags: map[string]string{
                "code": "500",
            },
        }})

        var receivedEnvelope []*loggregator_v2.Envelope
        Expect(mockWriter.WriteInput.Msg).To(Receive(&receivedEnvelope))
        Expect(receivedEnvelope).To(HaveLen(1))
        Expect(receivedEnvelope[0].GetCounter().GetTotal()).To(Equal(uint64(10)))

        Expect(mockWriter.WriteInput.Msg).To(Receive(&receivedEnvelope))
        Expect(receivedEnvelope).To(HaveLen(1))
        Expect(receivedEnvelope[0].GetCounter().GetTotal()).To(Equal(uint64(25)))

        Expect(mockWriter.WriteInput.Msg).To(Receive(&receivedEnvelope))
        Expect(receivedEnvelope).To(HaveLen(1))
        Expect(receivedEnvelope[0].GetCounter().GetTotal()).To(Equal(uint64(20)))
    })
```

**Comment by kitemongerer** [comment]
Additionally, it seems like origin is now in the Tags section rather than DeprecatedTags meaning counters from different origins may be aggregated together if they have the same name. See below envelope with empty DeprecatedTags:

```{"timestamp":"1536683581952216401","sourceId":"a5099325-9536-4b81-ba62-2826ad9a8462","instanceId":"0","deprecatedTags":{},"tags":{"applicationGuid":"a5099325-9536-4b81-ba62-2826ad9a8462","applicationInstance":"0","deployment":"prism","index":"2bb6b8cc-f202-4cde-83f9-dbed098ff15d","ip":"10.0.1.3","job":"prism_endpoint_worker","origin":"a5099325-9536-4b81-ba62-2826ad9a8462"},"gauge":{"metrics":{"go_gc_duration_seconds_sum":{"value":0.000591412}}}}```

**Comment by bradylove** [comment]
Thanks for reporting this issue, we have fixed it in a110caf.
